rotkehlchen/tests/db/test_db_upgrades.py

```diff
@@ -817,6 +820,20 @@ def try_insert_mapping(cur):
     with pytest.raises(sqlcipher.IntegrityError):  # pylint: disable=no-member
         with db_v34.conn.write_ctx() as write_cursor:
             try_insert_mapping(write_cursor)
+        cursor.execute('SELECT COUNT(*) from assets WHERE identifier=?', ('BIFI',))
+        assert cursor.fetchone() == (1,)
+        # check that assets are update correctly in the user db
+        cursor.execute(
+            'SELECT COUNT(*) from manually_tracked_balances WHERE asset=?',
+            ('BIFI',),
+        )
+        assert cursor.fetchone() == (1,)
+
+    with db_v34.user_write() as write_cursor:
+        # when opening the database with this version of develop we introduced the new assets
+        # in the database. that is not possible in the user database due to
+        # https://github.com/rotki/rotki/blob/develop/rotkehlchen/globaldb/upgrades/manager.py#L40
+        write_cursor.execute('DELETE FROM assets WHERE identifier LIKE "eip155:%"')
```

Comment:
What are you doing here? I don't understand. Also the comment makes no sense.

Why would you have a CAIP format identifier thing in the v34 DB?

Reply:
The reason is because in develop the new schema of the globaldb is valid and when this line gets executed https://github.com/rotki/rotki/blob/develop/rotkehlchen/db/dbhandler.py#L272 at the moment of opening the db before the migration the assets with the CAIPs format get copied

Reply:
Then you should solve this properly like we do in other upgrades which is to patch this function to do a no-op before and use it in this test